R/sensitivity_labels.R

```diff
@@ -143,19 +153,102 @@ apply_sensitivity_label <- function(file, label) {
     )
   }
 
-  # Check file is Excel workbook
-  file_extension <- tolower(tools::file_ext(file))
-  if (!file_extension %in% c("xlsx", "xls")) {
-    cli::cli_abort(
-      "{.arg file} must be an Excel workbook with {.val .xlsx} or {.val .xls} extension, not {.val .{file_extension}}."
-    )
+  # Check file is valid
+  file_ext <- tolower(tools::file_ext(file))
+  if (!file_ext %in% c("xlsx", "xls", "docx")) {
+    cli::cli_abort("{.arg file} must be an Excel workbook or Word document with{.val .xlsx}, {.val .xls}, or {.val .docx} extension, not {.val .{file_ext}}.")
   }
 
-  # Load workbook and apply label
-  wb <- openxlsx2::wb_load(file)
-  xml <- sensitivity_label_xml[[label]]
-  wb <- openxlsx2::wb_add_mips(wb, xml)
-  openxlsx2::wb_save(wb, file)
+  ## Apply label to Excel workbooks ----
+
+  if(file_ext %in% c("xlsx", "xls")){
+    wb <- openxlsx2::wb_load(file)
+    xml_map <- .get_sensitivity_xml_map()
+    xml <- xml_map[[label]]
+    wb <- openxlsx2::wb_add_mips(wb, xml)
+    openxlsx2::wb_save(wb, file)
+  }
+
+  ## Apply label to Word docs ----
+
+  if(file_ext == "docx"){
+    # Parsing input
+    file_path <- dirname(file)
+    file_name <- basename(file)
+    file_type <- tools::file_ext(file)
+    # Removing file type from actual name
+    file_name <- sub(paste0(".", file_type), "", file_name)
+
+    # Zipping process needs its own directory
+    # We'll reset it using on.exit later
+    current_wd <- getwd()
```

Comment:
I need to check this approach works when creating new documents using PHS Rmd templates - I'm hoping to add an option to allow setting the sensitivity label when you run the function to start e.g. a new annual report from template but if it can't be set up at the beginning, we can still use this function to post-process.

Reply:
Thanks, I have an idea of how to implement using your function (at least for the `phs_report_docx` Rmd type) but will maybe merge your changes in first and then open a new PR for that work.